types/react-redux/index.d.ts

```diff
@@ -171,7 +171,7 @@ export interface Connect {
 
     <no_state = {}, TDispatchProps = {}, TOwnProps = {}>(
         mapStateToProps: null | undefined,
-        mapDispatchToProps: TDispatchProps,
+        mapDispatchToProps: MapDispatchToPropsParam<TDispatchProps, TOwnProps>,
```

Comment:
Without this, `connect` would use the whole `MapDispatchToProps` (_union_) for the `TDispatchProps` (object) generic.